server/src/main/java/org/opensearch/index/codec/CodecService.java

```diff
@@ -104,4 +105,10 @@ public Codec codec(String name) {
     public String[] availableCodecs() {
         return codecs.keySet().toArray(new String[0]);
     }
+
+    public static boolean isSupportedSetting(String codec, Setting setting) {
```

Comment:
This is not going to work for the reasons explained before: static methods are not overridable

Reply:
makes sense!